Testing/C2DTests.cs

```diff
@@ -68,17 +75,64 @@ public async Task SendD2CWithRetainFlagTrue()
 
             var cancellationSource = new CancellationTokenSource();
             cancellationSource.CancelAfter(3000);
-            var messages = await receiverConsumer.ReceiveMessagesFromDeviceAsync(cancellationSource.Token);
+            var messages = await receiver.ReceiveMessagesFromDeviceAsync(cancellationSource.Token);
 
             // Assert - verify message was received + mqtt-retain set to true
             var sentMessage = messages.FirstOrDefault(x => x.Payload == payload);
             Assert.IsTrue(sentMessage != null && sentMessage.RetainFlag == "true");
         }
 
         [Test]
+        public async Task ReceiveAllC2DMessagesWithCleanSessionFalse()
```

Comment:
This test is very flakey. It's less flakey with the sleep on line 107.

I think this means that IoTHub is failing at this MQTT feature. What do you think @vjrantal? I'll also bring this up in the parking lot on monday.

Reply:
Code looks good to me, let's check the reason for the flakiness on Monday.